### PR TITLE
Introduce communication channel from workflow-controller to executor through pod annotations

### PR DESCRIPTION
**Pull request by jessesuen**
This change creates a downward communication channel from the controller to executor through
the use of pod annotations. Its initial use will be to terminate daemoned containers by
updating 'ExecutionControl' parameters to the executor. In this case, it updates the deadline
value to be a zero-value timestamp. The executor has been modified to now monitor updates to the annotations and will obey the updated deadline, allowing deadlines to be set/updated/unset after initial start.

In the future, ExecutionControl can be extended to support things like workflow, step, dag
level timeouts, as well as the ability to to cancel a workflow without deleting pods.

Resolves issue #499.
Partially implements support for #527.

### CODE REVIEW

**Comment by jessesuen** [comment]
The contents of the new file steps.go is code which moved from operation.go but with tweaked killDeamonedChildren to use the new method updateExecutionControl().

File: examples/hello-world.yaml [hunk 1]
```diff
@@ -8,5 +8,5 @@ spec:
   - name: whalesay
     container:
       image: docker/whalesay:latest
-      command: [cowsay]
-      args: ["hello world"]
+      command: [sh, -c]
+      args: ["sleep 30"]
```
Comment (shrinandj) [inline examples/hello-world.yaml]: Why is this needed?

Comment (jessesuen) [reply examples/hello-world.yaml]: This is a mistake. Thanks for catching this.

Comment (jessesuen) [reply examples/hello-world.yaml]: Fixed.
